pkg/authorization/apiserver/registry/rolebindingrestriction/etcd/etcd.go

```diff
@@ -1,43 +1,68 @@
 package etcd
 
 import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/watch"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	metainternalversion "k8s.io/apimachinery/pkg/apis/meta/internalversion"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apiserver/pkg/registry/generic"
-	"k8s.io/apiserver/pkg/registry/generic/registry"
 	"k8s.io/apiserver/pkg/registry/rest"
-	"k8s.io/kubernetes/pkg/printers"
-	printerstorage "k8s.io/kubernetes/pkg/printers/storage"
 
-	authorization "github.com/openshift/api/authorization"
 	authorizationapi "github.com/openshift/origin/pkg/authorization/apis/authorization"
-	"github.com/openshift/origin/pkg/authorization/apiserver/registry/rolebindingrestriction"
-	printersinternal "github.com/openshift/origin/pkg/printers/internalversion"
 )
 
 type REST struct {
-	*registry.Store
 }
 
 var _ rest.StandardStorage = &REST{}
+var _ rest.Scoper = &REST{}
 
 // NewREST returns a RESTStorage object that will work against nodes.
-func NewREST(optsGetter generic.RESTOptionsGetter) (*REST, error) {
-	store := &registry.Store{
-		NewFunc:                  func() runtime.Object { return &authorizationapi.RoleBindingRestriction{} },
-		NewListFunc:              func() runtime.Object { return &authorizationapi.RoleBindingRestrictionList{} },
-		DefaultQualifiedResource: authorization.Resource("rolebindingrestrictions"),
+func NewREST() (*REST, error) {
+	return &REST{}, nil
+}
+
+func (r *REST) NamespaceScoped() bool {
+	return false
+}
 
-		TableConvertor: printerstorage.TableConvertor{TablePrinter: printers.NewTablePrinter().With(printersinternal.AddHandlers)},
+func (r *REST) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
+	return nil, errors.NewInternalError(fmt.Errorf("unsupported"))
```

Comment:
All of the `unsupported` bits look like something we can create a struct for and embed.